soc/xtensa/intel_adsp/common/include/cavs-ipc-regs.h

```diff
@@ -43,23 +45,28 @@
  *    This is historically unused and unsupported by simulation tools.
  *
  * Note that while the scheme in #1 may look like an atomic message,
- * there is no queuing in the hardware!  This is just a regular level
+ * there is no queuing in the hardware! This is just a regular level
  * triggered interrupt with some scratch registers on which multiple
- * contexts can race.  The design intent is that the sender polls the
+ * contexts can race. The design intent is that the sender polls the
  * IDR.BUSY bit (or waits for a return interrupt via #2) to be sure
- * the link is available.  And even then two independent initiators
+ * the link is available. And even then two independent initiators
  * (cores, threads, whatever) trying to initiate a message may race if
- * they have access to the same endpoint.  Use with care.
+ * they have access to the same endpoint. Use with care.
  *
  * Note also that the ancestral cAVS 1.5 version of this interface is
  * very similar (the registers had different names, but that's been
  * unified here), but with a different layout and with DONE signaled
  * via bit 30 of the TDD register when BUSY is cleared in TDR on the
- * other side and not via a separate TDA/IDA.  This means that it's
+ * other side and not via a separate TDA/IDA. This means that it's
  * not possible to defer completion of a message on cAVS 1.5 as the
  * interrupt would remain active.
  */
-struct cavs_ipc {
+#if defined(CONFIG_INTEL_ADSP_CAVS)
```

Comment:
if this header is CAVS only, it should not be in common/include. Also, instead of having the structure ifdefed this way (v15 vs the rest) it would be nicer and cleaner to just have a header per platform and put it in `cavs/include/intel_<platform>_adsp/` instead where the `intel_adsp_ipc` has the members for the relevant platform only.